src/Auth0.php

```diff
@@ -247,8 +266,18 @@ public function login($state = null, $connection = null) {
       $params['scope'] = $this->scope;
     }
 
+    if($state === null) {
+      $state = $this->stateHandler->issue();
+    } else {
+      $this->stateHandler->store($state);
+    }
+
     $params['response_mode'] = $this->response_mode;
 
+    if($additional_params) {
```

Comment:
if my statement is correct, then this will always be either `[]` or some (I say good) value passed by the user. In any case, this if is always `true`. Maybe the check should be `if($additional_params && $additional_params.length/size>0)`

Reply:
That `if` will check for truth-y values so an empty array or `null` will skip it. That said, if someone passed in a `1` or a string or something, they'll get an error. Probably should be something like:

```php
if(!empty($additional_params) && is_array($additional_params)) {
      $params = array_replace($params, $additional_params);
    }
```

Reply:
I will accept the change, however as long as the parameter type is documented somewhere, users shouldn't be passing invalid (non-array) values. So next time: doc + `if($var)` should be enough.